pkg/p2p/discover.go

```diff
@@ -14,7 +14,7 @@ import (
 	madns "github.com/multiformats/go-multiaddr-dns"
 )
 
-func Discover(ctx context.Context, addr ma.Multiaddr, f func(ma.Multiaddr) (stop bool, err error)) (stopped bool, err error) {
+func Discover(ctx context.Context, addr ma.Multiaddr, f func(ma.Multiaddr) (bool, error)) (bool, error) {
```

Comment:
would be nice at some point to clean this signature up... not relevant for this PR

Reply:
@acud would you create the issue with that to clean up, so that we do not forget?

Reply:
Oups, sorry, did not see this.. what should we clean-up there? Do you think I should bring back the named parameters? I did not like cuz they are used inside in some checks and I did not like them initializxed before

Reply:
In general I am fine with removing the named parameters. In this case they were providing some form of documentation, as the bools were described with them. But I have nothing agains that you removed them and only line in the comment to describe them would be nice.